docs/2.8.0/docs/daml/reference/choices.rst

```diff
@@ -66,21 +27,30 @@ Controllers
 
 .. literalinclude:: ../code-snippets/Reference.daml
    :language: daml
-   :start-after: -- start choice-first controller snippet
-   :end-before: -- end choice-first controller snippet
-   :caption: Option 1 for specifying choices: choice name first
-
-.. literalinclude:: ../code-snippets/Reference.daml
-   :language: daml
-   :start-after: -- start controller-first controller snippet
-   :end-before: -- end controller-first controller snippet
-   :caption: Option 2 for specifying choices (deprecated syntax): controller first
+   :start-after: -- start new controller snippet
+   :end-before: -- end new controller snippet
 
 - ``controller`` keyword
 - The controller is a comma-separated list of values, where each value is either a party or a collection of parties.
 
   The conjunction of **all** the parties are required to authorize when this choice is exercised.
 
+.. warning::
+
+  You **must** make sure that the controller parties are observers (or signatories) of the contract, otherwise they won't be able to see the contract (and therefore won't be able to exercise the choice).
```

Comment:
```suggestion
  You **must** make sure that the controller parties are observers (or signatories) of the contract, otherwise they cannot see the contract (and therefore cannot exercise the choice).
```